extensions/amp-timeago/0.1/locales.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This file registers all necessary locales supported by amp-timeago.
+ */
+
+import {format, register} from 'timeago.js/dist/timeago.full.min.js';
+import cs from 'timeago.js/lib/lang/cs';
+import da from 'timeago.js/lib/lang/da';
+import ka from 'timeago.js/lib/lang/ka';
+import oc from 'timeago.js/lib/lang/oc';
+
+export {format};
+
+/**
+ * timeago.full.min.js only contains the following 41 locales:
+ * ar, be, bg, bn_IN, ca, de, el, en_short, en_US, es, eu, fa, fi, fr, gl, he,
+ * hi_IN, hu, id_ID, it, ja, ko, ml, my, nb_NO, nl, nn_NO, pl, pt_BR, ro, ru,
+ * sq, sr, sv, ta, th, tr, uk, vi, zh_CN, zh_TW
+ *
+ * Register the rest to create full support for the 46 languages
+ * provided by the timeago.js library.
+ *
+ * TODO(wg-components): These can be removed once all the languages are
+ * exported in timeago.js. See https://github.com/hustcc/timeago.js/issues/238.
+ */
+register('cs', cs);
+register('da', da);
+register('ka', ka);
+register('oc', oc);
+
+/**
+ * timeago.full.min.js registers some locales in a different format than what
+ * is currently supported by amp-timeago. For backwards compatibility, the
+ * following additional locale formats are converted to their registered type.
+ * @param {string} locale
+ * @return {string}
+ */
+export function getLocale(locale) {
```

Comment:
Should we be doing this in JS or processing the locale information in a publishing step from the origin library?

Reply:
If the latter, I believe we would need an I2D on the formats converted here and keep this as an intermediary step. Unless I am misunderstanding what you are suggesting?

Reply:
Suggestion was to remove the mapping here and generate the output from the library output as a build step.

Reply:
Oh got it, done.

Reply:
Re-opening this thread. After struggling with adding to `sources.js` and chatting with @samouri, the `patch` approach during build isn't compatible with importing the minified modules. He recommended in [3cd6067](https://github.com/ampproject/amphtml/pull/32809/commits/3cd6067541d78c95aa8ef8addeee271c582f93a1) to keep this in JS to most clearly get `gulp dist` to build

Reply:
A couple of things:
1. It would be nontrivial to point closure compiler to either the `esm` or `lib` folders of timeago and have it work correctly. depending on which folder you try to point to, something explodes. (e.g. expecting all the es5 imports to have `.default` after their import etc). For that reason, using the files within the prebuilt dist folder is the only feasible option unless we move off Closure.
2.  Since our locale format doesn't match up with timeago, we need to have the conversion somewhere. The _absolutely_ ideal location is of course for only a single mapping layer to exist, but to patch the pre-minified file would be difficult and brittle (liable to break at any package update)

Reply:
Going to merge this for now. If folks have any recommended changes from this discussion happy to follow up!